### PR TITLE
feat: Social links collection detail

### LINKED ISSUE #5226: Social links collection detail
image

- we are going to group the social links of the collection 
- if the owner won't connect any social link, the icons won't be displayed 
- there is still the little separator, that is there until at least one social link for the collection is there 
- right side remains the same 
- main thing why we are doing this is mobile version 

### mobile
image

- social links are not over the banner anymore
- the are separated, social links left, more options/share right 
- this way the name can expand and there is more whitespace

thank you!

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #5226
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

/bsx/collection/1865909717?redesign=true

image

image

### CODE REVIEW

File: components/collection/HeroButtons.vue [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+      class="hero-buttons is-flex is-justify-content-flex-start is-align-items-end px-2">
+      <div class="is-flex">
+        <a
+          v-if="twitter"
+          :href="`https://twitter.com/${twitter}`"
+          target="_blank">
+          <NeoButton icon="twitter" icon-pack="fab" class="square-32" />
+        </a>
+        <a
+          v-if="twitter"
```
Comment (roiLeo) [inline components/collection/HeroButtons.vue:13]:
```suggestion
          v-if="instagram"
```

File: components/collection/HeroButtons.vue [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+          :href="`https://instagram.com/${instagram}`"
+          target="_blank">
+          <NeoButton
+            v-if="instagram"
```
Comment (roiLeo) [inline components/collection/HeroButtons.vue:17]:
```suggestion
```

File: components/collection/HeroButtons.vue [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            icon="instagram"
+            icon-pack="fab"
+            class="square-32" />
+        </a>
+        <a
+          v-if="twitter"
```
Comment (roiLeo) [inline components/collection/HeroButtons.vue:23]:
```suggestion
          v-if="discord"
```

File: components/collection/HeroButtons.vue [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+          :href="`https://discord.com/users/${discord}`"
+          target="_blank">
+          <NeoButton
+            v-if="discord"
```
Comment (roiLeo) [inline components/collection/HeroButtons.vue:27]:
```suggestion
```

**Review by roiLeo — COMMENTED** [review]
Could you change relative path?
(and check deepsource `rel="noopener noreferrer"`)

**Comment by exezbcz** [comment]
aaand how does it look like when there are more collection links? 👀 
otherwise look great!

File: components/collection/HeroButtons.vue [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+  width: 32px;
+  height: 32px;
+}
+.vertical-seperator {
+  height: 24px;
+  margin-bottom: 4px;
+  width: 1px;
+  @include ktheme() {
+    background-color: theme('border-color');
+  }
```
Comment (vikiival) [inline components/collection/HeroButtons.vue:167]: can we avoid pxs?

**Review by vikiival — CHANGES_REQUESTED** [review]
Also agree with @roiLeo

File: components/collection/HeroButtons.vue (hunk 5 above)
Comment (roiLeo) [reply components/collection/HeroButtons.vue:167]: for square-32 I don't see problem to use px, but can't we reuse same buttons component as the one in footer?

Comment (Jarsen136) [reply components/collection/HeroButtons.vue:162]: I just take a look at the components in footer and found that there are lots of custom style and tag like `<li> <a> <b-icon> ` : )  We use `NeoButton` here would be a better choice.
image

Comment (roiLeo) [reply components/collection/HeroButtons.vue:162]: Aight leave it like this for now

**Comment by Jarsen136** [comment]
> Could you change relative path?
(and check deepsource rel="noopener noreferrer")

✅ fixed

File: components/collection/CollectionInfo.vue [hunk 6]
```diff
@@ -48,6 +49,8 @@
 import CollectionInfoLine from './collectionInfoLine.vue'
 import CommonTokenMoney from '@/components/shared/CommonTokenMoney.vue'
 import IdentityIndex from '@/components/identity/IdentityIndex.vue'
+import HeroButtons from './HeroButtons.vue'
```
Comment (roiLeo) [inline components/collection/CollectionInfo.vue:52]:
```suggestion
import HeroButtons from '@/components/collection/HeroButtons.vue'
```

**Review by roiLeo — APPROVED** [review]
What I see that can be improved:
- don't wrap `button` with `<a />` tag
- `NeoButon` variant (only icon & size or `is-32x32`?)
- replace `b-modal` with oruga equivalent
- buefy toast => oruga notification (maybe `NeoToast`)

otherwise lgtm ✅

**Comment by Jarsen136** [comment]
> What I see that can be improved:
> 
> * don't wrap `button` with `<a />` tag

✅  I have replaced `<a/>` tag with window.open
> * `NeoButon` variant (only icon & size or `is-32x32`?)

 `is-32x32` only works for image tag. I'm not about to add a variant only for this size case.

> * replace `b-modal` with oruga equivalent
> * buefy toast => oruga notification (maybe `NeoToast`)
> 
Yes, that could be another common issue related with buefy component migration
- replace `b-modal` with `o-modal`  https://oruga.io/components/modal.html
- buefy toast => oruga notification (maybe `NeoToast`)

**Comment by vikiival** [comment]
@exezbcz 🤍

**Review by vikiival — APPROVED** [review]

**Comment by vikiival** [comment]
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 33.73 USD/KSM ~ 0.889 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0x8850ee5be90db80a25aee7293a6794b29ed25d1e8426688f0c78e7c52f4480a0](https://kusama.subscan.io/extrinsic/0x8850ee5be90db80a25aee7293a6794b29ed25d1e8426688f0c78e7c52f4480a0)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**